### ISSUE TITLE
len: `Len` shouldn't be recommended when it's the expected value

### ISSUE THREAD

**Issue by dhrubabasu** [Antonboom/testifylint#9]
IMO this should only flag if `len(foo)` is the *actual* val, not the expected val.

These should not be flagged:
- `require.Equal(len(foo), 2)`
- `require.Equal(t, len(foo), 2)`

These should be flagged:
- `require.Equal(2, len(foo))`
- `require.Equal(t, 2, len(foo))`

Currently they are all flagged.

**Comment by Antonboom** [comment]
Hi! Thank you for report.

Could you provide examples with variables, not constants?
Because your examples looks strange.

Or just confirm my thoughts:

```go
require.Equal(t, len(foo), 2)    // Obvious mistake, expected-actual checker mark it
-> require.Equal(t, 2, len(foo))
-> require.Len(t, foo, 2)        // We just skip the previous step now, because len checker has more priority 
                                 // than expected-actual
```

```go
require.Equal(t, len(foo), expectedValue) // The similar situation
```

```go
require.Equal(t, len(foo), someVarialbe) // Should be ignored by expected-actual and len
```

```go
require.Equal(t, someVarialbe, len(foo))
-> require.Len(t, len(foo), someVarialbe)
```

```go
require.Equal(t, len(bar), len(foo))    // Should be ignored by len
```

**Comment by dhrubabasu** [comment]
Those examples were all `require.Equal(expected, actual)`. With your `expected-actual` checker, I think what you have is probably the approach that'll work for most golang projects. If somebody is doing `require.Equal(t, len(foo), 2))`, then explicitly nolinting it is ok.

I think you had a typo for one of them, should remove `len` like this:
```
require.Equal(t, someVarialbe, len(foo))
-> require.Len(t, foo, someVarialbe)
```

Rest LGTM

Side-note: I wrote a bash script with a ton of small rules like this so I'm very happy that you made this package ❤️

**Comment by Antonboom** [comment]
@dhrubabasu, fixed. Thanks!

Now `len` supports this
```go
require.Equal(t, len(foo), someVarialbe)  // Should be ignored by expected-actual and len
```

And also ignores expressions like `assert.True(len(foo) == someVarialbe)` <br> (when the second operand is not int literal).
